public/display.js

```diff
@@ -1018,35 +1018,55 @@ function displayActivity() {
   if (typeof callback === 'function') {
    callback()
   }
-  lastMessages += news.data.map((newsMessage) =>
-   attachNewsMessage(element, newsMessage)
+  lastMessages.push(
+   ...news.data.map((newsMessage) =>
+    attachNewsMessage(element, newsMessage)
+   )
   )
+  filterAgain()
+  return nextChunk
  }
 
  function clear() {
   element.innerHTML = ''
   hide()
  }
 
- function filter(filterText) {
-  const terms = filterText
+ let filterTerms = []
+ let numFilteredNews = 0
+ async function filter(filterText) {
+  filterTerms = filterText
    .split(/\s+/)
    .map((x) => x.toLowerCase())
+   .filter((x) => x !== '')
+  filterAgain()
+  while (
+   numFilteredNews < 1 &&
+   (await withLoading(loadMore())) > -1
+  ) {
+   // Non-op
+  }
+ }
+
+ function filterAgain() {
+  numFilteredNews = 0
   for (const {
    channel,
    element,
    message,
    parentMessage,
   } of lastMessages) {
    const messageIncludesAllTerms =
-    terms.length === 0
+    filterTerms.length === 0
      ? true
-     : terms.every(
+     : filterTerms.every(
         (term) =>
          channel.includes(term) ||
          message.includes(term) ||
          parentMessage?.includes(term)
        )
+   if (messageIncludesAllTerms)
+    numFilteredNews++
```

Comment:
Won't need this with the solution to check scrolling again

Reply:
Removed.